File: ruby/lib/Sikaru/tool_providers/types/register_tool_provider_request.rb
# frozen_string_literal: true

module Sikaru
  module ToolProviders
    module Types
      class RegisterToolProviderRequest < Internal::Types::Model
        field :project_id, -> { String }, optional: false, nullable: false

        field :approval_policy_ref, -> { String }, optional: true, nullable: false

        field :broker_endpoint_ref, -> { String }, optional: false, nullable: false

        field :capability_prefix, -> { String }, optional: false, nullable: false

        field :credential_owner, -> { String }, optional: true, nullable: false

        field :display_name, -> { String }, optional: false, nullable: false

        field :provider_type, -> { String }, optional: false, nullable: false

        field :tool_catalog_ref, -> { String }, optional: false, nullable: false
      end
    end
  end
end

File: typescript/src/api/resources/toolProviders/client/requests/RegisterToolProviderRequest.ts

/**
 * @example
 *     {
 *         broker_endpoint_ref: "broker_endpoint_ref",
 *         capability_prefix: "capability_prefix",
 *         display_name: "display_name",
 *         provider_type: "provider_type",
 *         tool_catalog_ref: "tool_catalog_ref"
 *     }
 */
export interface RegisterToolProviderRequest {
    approval_policy_ref?: string | null;
    broker_endpoint_ref: string;
    capability_prefix: string;
    credential_owner?: string;
    display_name: string;
    provider_type: string;
    tool_catalog_ref: string;
}
